app/deed/server.py

```diff
@@ -70,13 +71,20 @@ def delete(id_):
         if deed is None:
             abort(status.HTTP_404_NOT_FOUND)
         else:
-            return jsonify(id=id_), status.HTTP_200_OK
+            return {'id': id_}, status.HTTP_200_OK
 
     @blueprint.route('/deed/<deed_id>/<borrower_id>/signature/',
                      methods=['POST'])
     def sign(deed_id, borrower_id):
+        def find_deed():
+            deed_ = Deed.get(deed_id)
+            if deed_ is None:
+                abort(status.HTTP_404_NOT_FOUND)
```

Comment:
early and hidden returns are a prime area for confusion. Can you simplify this function to remove the need for early returns? I'd remove this find_deed function and take the other functions (`sign_allowed` and `sign_deed`) in to the deed object so the flow goes:
```
deed = get deed
if deed is None
  return 404
else 
  try
    deed.sign
    return 200
  catch
    return 403